### ISSUE TITLE
[Question] Reconfiguring Traefik

### ISSUE THREAD

**Issue by skriems** [kube-hetzner/terraform-hcloud-kube-hetzner#155]
Hey, first of all, thank you for this repo! :)

While playing around with it, I observed some things I couldn't figure out on my own. Sorry I'm completely new to terraform. A few days ago I was able to specify a `cluster_name` and all went fine, but now with the latest version it fails with

```
╷
│ Error: name is already used (uniqueness_error)
│ 
│   with hcloud_placement_group.control_plane[0],
│   on main.tf line 50, in resource "hcloud_placement_group" "control_plane":
│   50: resource "hcloud_placement_group" "control_plane" {
│ 
╵
╷
│ Error: name is already used (uniqueness_error)
│ 
│   with hcloud_placement_group.agent[0],
│   on main.tf line 56, in resource "hcloud_placement_group" "agent":
│   56: resource "hcloud_placement_group" "agent" {
│ 
╵
```

So I went ahead without a cluster_name for now, no big deal.

But the crucial thing to me now is the ability to change the traefik config. I changed the `traefik_additional_options` and expected that an additional `terraform apply...` would do the trick, but that's not the case. So I copy/pasted the `kustomization_backup` file to `kustomization.yaml` and invoking `kubectl apply -k ./` fails with an error, basically saying that the `traefik_config.yaml` cannot be found. My next step is to copy/paste the traefik_config.yaml.tpl and continue from there, but it feels like this could be improved or is unexpected actually. In addition to `traefik_additional_options` I have set these values in `terraform.tfvars`

```
traefik_acme_tls   = true
traefik_acme_email = "<email>"
```

So basically my question ist - how am I supposed to reconfigure traefik without destroying the whole cluster :D

**Comment by mysticaltech** [comment]
Basically, to reconfigure Traefik, just use traefik_congig.yaml (copy from the template indeed) and edit. They apply with kubectl apply -f.

For the uniqueness name error, make sure your nodepools do not have the same names.

If you are sure it's related to the cluster name, let me know! Will check anyways.

**Comment by mysticaltech** [comment]
As for SSL, better go with cert-manager or Cloudfare. Please search the issues, we've discussed this in others threads already! ☺️

**Comment by skriems** [comment]
@mysticaltech thanks for your hint to cert-manager or cloudfare, I will look into this.

As for the cluster_name: I started from scratch with the cluster_name set and it failed. I have not altered any names of my nodepools

**Comment by mysticaltech** [comment]
Ok, thanks for the confirmation, I will investigate the cluster name issue!

**Comment by mysticaltech** [comment]
Hey @skriems, the cluster_name works well, just tried (see screenshot). The only thing is that it has to respect the same format as the domain name, only alphanumeric or/and dashes eventually, so other special characters are allowed.

Also, make sure that your different nodepools all have different names! Last but not least, do migrate to the latest version of the cluster.

![ksnip_20220420-093154](https://user-images.githubusercontent.com/518555/164175612-7c5f26aa-83f7-4fa5-abfd-49eadbb12448.png)

**Comment by mysticaltech** [comment]
Ah, it is also possible that you already had another cluster in your account with the same name! That would also cause the uniqueness error!

**Comment by skriems** [comment]
ok then that is definitely the cause :)
